browser/profiles/brave_renderer_updater.cc

```diff
@@ -235,12 +244,29 @@ void BraveRendererUpdater::UpdateRenderer(
       base::FeatureList::IsEnabled(playlist::features::kPlaylist) &&
       pref_service->GetBoolean(playlist::kPlaylistEnabledPref);
 
+  bool is_yt_hd_quality_playback_enabled =
+      IsYTHDQualityPlaybackEnabled(pref_service);
   (*renderer_configuration)
       ->SetConfiguration(brave::mojom::DynamicParams::New(
           install_window_brave_ethereum_provider,
           install_window_ethereum_provider,
           allow_overwrite_window_ethereum_provider,
           brave_use_native_solana_wallet,
           allow_overwrite_window_solana_provider, de_amp_enabled,
-          onion_only_in_tor_windows, widevine_enabled, playlist_enabled));
+          onion_only_in_tor_windows, widevine_enabled, playlist_enabled,
+          is_yt_hd_quality_playback_enabled));
+}
+
+bool BraveRendererUpdater::IsYTHDQualityPlaybackEnabled(
+    PrefService* pref_service) {
+#if BUILDFLAG(IS_ANDROID)
+  int qualityPref = pref_service->GetInteger(kYTVideoQualityPref);
+  return qualityPref == static_cast<int>(settings::YTVideoQuality::kOn) ||
```

Comment:
this is not something `BraveRendererUpdater` should have to understand. You should send the raw pref value and also this is not going to correctly update when the connection type changes

Reply:
we have to send value based on preference set by user in the settings and connection type and for that we need connection type only when site starts loading. we are not listening for connection type changes.